.github/workflows/publish.yml

```diff
@@ -0,0 +1,61 @@
+name: Publish to codemod registry
+
+on:
+  push:
+    branches: ["main"]
+    paths: ["recipes/**"]
+  pull_request:
+    branches: ["main"]
+    paths: ["recipes/**"]
+    types: [closed]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: dorny/paths-filter@v3
+        id: filter
+        name: Filter codemods
+        with:
+          list-files: shell
+          filters: |
+            codemods:
+              - '**/.codemodrc.json'
+
+      - name: export files
+        run: |
+          echo "Modified files: ${{steps.filter.outputs.codemods_files}}"
+          echo "CODEMOD_FILES=${{steps.filter.outputs.codemods_files}}" >> $GITHUB_ENV
+          echo "Modified status: ${{steps.filter.outputs.codemods}}"
+          echo "CODEMOD_STATUS=${{steps.filter.outputs.codemods}}" >> $GITHUB_ENV
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: node
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci --include="optional"
+
+      - name: Generate bundle & publish to codemod registry
+        run: |
+          if [ -n "$CODEMOD_STATUS" ]; then
+            echo "Modified files: $CODEMOD_FILES"
+            ROOT_DIR=$(pwd)
+            for FILE in $CODEMOD_FILES; do
+              DIR=$(dirname "$FILE")
+              echo "Checking codemod: $DIR"
+              cd "$ROOT_DIR/$DIR"
+              npx esbuild --bundle --platform=node --outfile=bundle.js
+              npx codemod publish
+              echo "Codemod published"
+            done
+          else
+            echo "No codemods found"
+          fi
```

Comment:
What is we use a external script `.sh` or `.js` to allow dev to "simulate" a publish ?

Reply:
Hmm?